exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -106,11 +105,19 @@
   String RETURN_ERROR_FOR_FAILURE_IN_CANCELLED_FRAGMENTS =
       "drill.exec.debug.return_error_for_failure_in_cancelled_fragments";
 
+  String CLIENT_SUPPORT_COMPLEX_TYPES = "drill.client.supports-complex-types";
 
+  /**
+   * Configuration properties connected with dynamic UDFs support
+   */
+  String UDF_RETRY_TIMES = "drill.exec.udf.retry.times";
+  String UDF_DIRECTORY_STAGING = "drill.exec.udf.directory.staging";
+  String UDF_DIRECTORY_REGISTRY = "drill.exec.udf.directory.registry";
+  String UDF_DIRECTORY_TMP = "drill.exec.udf.directory.tmp";
+  String UDF_DIRECTORY_BASE = "drill.exec.udf.directory.base";
+  String UDF_DIRECTORY_FS = "drill.exec.udf.directory.fs";
```

Comment:
Let's discuss. DoY uses DFS to store files. Let's define a general property drill.dfs.home that points to the root DFS home folder. This can default to "/user/drill" for HDFS. ("/user" is an HDFS standard.) Users can change the root in a single place. Then, when merging DoY, we'll use the new drill.dfs.home property in place of the one that DoY currently uses.

Note also that Typesafe allows property values of the form:

drill.exec.udf.directory.fs: "$drill.dfs.home/udf" to make it easy to compute derived paths.